packages/opentelemetry-api/src/api/propagation.ts

```diff
@@ -53,19 +54,29 @@ export class PropagationAPI {
    * Inject context into a carrier to be propagated inter-process
    *
    * @param carrier carrier to inject context into
+   * @param setter Function used to set values on the carrier
    * @param context Context carrying tracing data to inject. Defaults to the currently active context.
    */
-  public inject(carrier: Carrier, context = contextApi.active()): void {
-    return this._propagator.inject(context, carrier);
+  public inject<Carrier>(
```

Comment:
why the order of params has changed ? I see later the order is correct (carrier, context, setter)

Reply:
It is based on which arguments are most likely to be omitted. The most common use case is `extract(carrier)`, the second most common is `extract(carrier, getter)`, the third most common is `extract(carrier, getter, context)`.

The order of the actual propagator itself has context first because that was the order before. Originally, before the global API, it was `inject(spanContext, format, carrier)`. Format has been removed, but the other order was preserved.

Reply:
Hmm I'm a bit confused with this as for for interface `HttpTextFormat` the order is (carrier, context, setter/getter), maybe I'm wrong but my understanding is that it should be the same in both cases ?

Reply:
One is the order of the public API, which I have ordered for the reasons above (some arguments optional and unlikely to be used). The other is the order of the interface, which accepts all arguments. I kept the order of the interface for now, but it can be made consistent in a future PR if we want.